atlasdb-api/src/main/java/com/palantir/atlasdb/transaction/api/TransactionManager.java

```diff
@@ -102,4 +102,9 @@
      * @return value returned by task
      */
     <T, E extends Exception> T runTaskReadOnly(TransactionTask<T, E> task) throws E;
+
+    /**
+     * Performs non-destructive cleanup when the transaction manager is no longer needed.
+     */
+    void close();
```

Comment:
That's exactly the point and why the fail fast sanity check is needed -- if txn manager has been closed, further interactions with it should throw as they aren't valid, and should not be allowed to trigger interactions that partially complete.

TransactionManager interface did not previously have a close method, and this change adds it, which is a back compat break in itself because we are not defining a default close implementation.